apps/builder/app/builder/features/seo/project-redirects.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Grid,
+  Separator,
+  Label,
+  Flex,
+  InputField,
+  Button,
+  Text,
+  theme,
+  List,
+  ListItem,
+  css,
+  SmallIconButton,
+} from "@webstudio-is/design-system";
+import { TrashIcon } from "@webstudio-is/icons";
+import { useState, type ChangeEvent } from "react";
+import type { ProjectSettings } from "./project-settings";
+
+const redirectListStyle = css({
+  maxHeight: theme.spacing[22],
+  paddingLeft: 0,
+  overflowY: "scroll",
+  listStyle: "none",
+});
+
+export const ProjectRedirectionSettings = (props: {
+  settings: ProjectSettings;
+  onSettingsChange: (settings: ProjectSettings) => void;
+}) => {
+  const [oldPath, setOldPath] = useState<string | undefined>(undefined);
+  const [newPath, setNewPath] = useState<string | undefined>(undefined);
+  const [isValidOldPath, setIsValidOldPath] = useState<boolean>(true);
+  const [isValidNewPath, setIsValidNewPath] = useState<boolean>(true);
+
+  const redirects = props.settings?.redirects ?? {};
+  const redirectKeys = Object.keys(redirects);
+  const isValidRedirects =
+    oldPath !== undefined &&
+    isValidOldPath &&
+    newPath !== undefined &&
+    isValidNewPath;
+
+  const validateOldPath = () => {
+    if (oldPath === undefined) {
+      setIsValidOldPath(true);
+      return;
+    }
+    setIsValidOldPath(oldPath.charAt(0) === "/");
```

Comment:
ideally tihs validation is part of the schema, e.g. see how we refine Page path from validation, this way you can have both client-side and also server-side validation and its all in one place, because you will likely add more things for validations over time